src/main/java/com/thirdcc/webapp/service/impl/EventAttendeeServiceImpl.java

```diff
@@ -95,17 +107,70 @@ public EventAttendeeDTO save(EventAttendeeDTO eventAttendeeDTO) {
         return eventAttendeeRepository.findAll(pageable)
             .map(eventAttendeeMapper::toDto);
     }
-
+    
+    /**
+     * Get all the eventAttendees from an Event via Event Id.
+     *
+     * @param pageable the pagination information.
+     * @param eventId the eventId of the event
+     * @return the list of entities.
+     */
     @Override
     @Transactional(readOnly = true)
     public Page<EventAttendeeDTO> findAllByEventId(Pageable pageable, Long eventId) {
         log.debug("Request to get all EventAttendee by Event Id: {}", eventId);
         eventRepository
             .findById(eventId)
             .orElseThrow(() -> new BadRequestException("Event not found"));
-
-        return eventAttendeeRepository.findAllByEventId(pageable, eventId)
-            .map(eventAttendeeMapper::toDto);
+        List<EventAttendeeDTO> list = eventAttendeeRepository.findAllByEventId(eventId)
+                .stream()
+                .map(eventAttendeeMapper::toDto)
+                .map(this::mapEventAttendeeDetails)
+                .collect(Collectors.toList());
+        String orderProperty = null;
+        Direction orderDirection = null;
+        for(Sort.Order order : pageable.getSort()){
+            orderProperty = order.getProperty();
+            orderDirection = order.getDirection();
+        }
```

Comment:
loop the Sort and only get the last `orderProperty` & `orderDirection`

Reply:
Yes, actually currently front end only can set 1 sort at a time, except the user add other sort in url. In this way when the front end pass the sort only will get into the loop, so do not need to manually check for null value

Reply:
@kachun333 or i should change the code to this?

Iterator<Sort.Order> sortIterator = pageable.getSort().iterator();
        if(sortIterator.hasNext()){
            Sort.Order order = sortIterator.next();
            orderProperty = order.getProperty();
            orderDirection = order.getDirection();
        }

Reply:
@CarlosSia , yes, I think it is better 😄

Reply:
Resolve by https://github.com/Five-Fishes/Club-Management-Spring-Server/pull/97/commits/441581d9c802195494b4d9a359e315259ab7c1c7